packages/terra-navigation-layout/src/routingUtils.js

```diff
@@ -0,0 +1,105 @@
+import { matchPath } from 'react-router-dom';
+
+/**
+ * Given a route configuration and size, returns an Array of ordered routing data objects
+ * representing the routes available at that size.
+ * @param {Object} routeConfig The route configuration to reduce. The expected shape matches that of the routeConfigPropType.
+ * @param {String} size The breakpoint size used during configuration parsing.
+ * @param {Array[String]} parentPaths The array of String paths that are ancestors to the given routeConfig.
+ */
+const reduceRouteConfig = (routeConfig, size) => {
```

Comment:
Might be good to have some concrete examples of what is going on here in the comments for posterity.